### DISCUSSION TITLE
Was the last release of webpack supposed to fix the issue that is causing Alphabetical sort in production Storybook?

### DISCUSSION THREAD

**Discussion by catdc2017** [webpack/webpack#15628] [General]
I currently updated the webpack to the latest version.  I am now on 5.71 using Storybook 6 and Angular 13 when I export my SB stories webpack is re sorting them into Alphabetical order in production?  Was a fix for this released on Friday?

**Comment by vankop** [comment]
hm.. interesting, technically webpack sorts es imports alphabetically since 5.70 https://github.com/webpack/webpack/releases/tag/v5.70.0

______

but I'm not sure that this is the cause.

**Reply by catdc2017** [reply]
I presume that this is incorrect behaviour from this thread I found https://github.com/storybookjs/storybook/issues/15574 online.  

In storybook the sort order is decided by the position of the export.  When I build this locally it works perfectly but when I build it in production it sorts all stories / navigation in Alphabetical order and really strangely it is ignoring any numbers or special characters (I was trying to push API and Guidelines docs to the end of the list) but as I said it ignores standard ASCII sorting rules?   

Is that expected behaviour with webpack sort to ignore special characters, numbers or upper / lower case letters?

**Comment by catdc2017** [comment]
This is how navigation links are being displayed in production.  

<img width="176" alt="Screenshot 2022-04-05 at 10 57 33" src="https://user-images.githubusercontent.com/26303634/161729583-a8aa8074-05d1-447a-80a2-09074af677a7.png">

It is not respecting numbers / special characters.  

When I try to manually storySort: { } in preview.js it will only sort on root/folder/folder 

        storySort: {
            method: 'alphanumeric',
            order: [
                'Getting Started',
                ['Installation', 'Upgrading', 'Migrating from UITK4'],
                'Foundations',
                'Components',
                ['Autocomplete', ['Template', 'Reactive']],
            ],
            locales: '',
        },

**Reply by vankop** [reply]
please ask this in storybook issue tracker.. there is nothing todo from webpack side.

**Reply by catdc2017** [reply]
Ok thanks for the reply I will post it in Storybook .... but I think from reading online forums regarding this issue it is a webpack5 sort issue!  

As you can see this is definitely a problem.

**Reply by vankop** [reply]
import specifiers should be sorted alphabetically by esm spec..
so in case
a.js
```js
export const b = 2;
export const a = 1;
```
b.js
```js
export * as b from "./a"
```

index.js
```js
import {b} from "./b"
console.log(Object.keys(b)) // should print ['a','b']
```

**Reply by catdc2017** [reply]
stories.ts file structure below:

<img width="587" alt="Screenshot 2022-04-05 at 12 02 08" src="https://user-images.githubusercontent.com/26303634/161740382-ab5ac8d3-7fe7-4f67-b7c8-4743b6770061.png">

Should build as (which this is my local dev and it builds correctly)

<img width="245" alt="Screenshot 2022-04-05 at 12 02 44" src="https://user-images.githubusercontent.com/26303634/161740474-45629150-5ea8-43ef-9ab4-6311d2dfa3dd.png">

I have posted the issue in SB as well so I will see what they come back with.  Unfortunately I cannot seem to force a way to override the automatic alphabetic sorting : (    It feels very strange it is ignoring numbers as well??

Thanks for your time / help.  If SB comes back with a solution I will post it on this discussion as well as it may help someone else : )

**For added info this was working with Storybook and webpack 4 before.
